packages/common/src/codegen/utils/contractToInterface.ts

```diff
@@ -55,55 +70,81 @@ export function contractToInterface(
     throw new MUDError(`Contract not found: ${contractName}`);
   }
 
-  visit(contractNode, {
-    FunctionDefinition({
-      name,
-      visibility,
-      parameters,
-      stateMutability,
-      returnParameters,
-      isConstructor,
-      isFallback,
-      isReceiveEther,
-    }) {
-      try {
-        // skip constructor and fallbacks
-        if (isConstructor || isFallback || isReceiveEther) return;
-        // forbid default visibility (this check might be unnecessary, modern solidity already disallows this)
-        if (visibility === "default") throw new MUDError(`Visibility is not specified`);
-
-        if (visibility === "external" || visibility === "public") {
-          functions.push({
-            name: name === null ? "" : name,
-            parameters: parameters.map(parseParameter),
-            stateMutability: stateMutability || "",
-            returnParameters: returnParameters === null ? [] : returnParameters.map(parseParameter),
-          });
-
-          for (const { typeName } of parameters.concat(returnParameters ?? [])) {
-            const symbols = typeNameToSymbols(typeName);
-            symbolImports = symbolImports.concat(symbolsToImports(ast, symbols, findInheritedSymbol, qualifiedSymbols));
+  for (const match of contractNode.query([
+    Query.create(`
+      @function [FunctionDefinition [FunctionName [Identifier]]]
+    `),
+    Query.create(`
+      @error [ErrorDefinition]
+    `),
+  ])) {
+    switch (match.queryIndex) {
+      case 0: {
+        // Functions
+        const funcNode = match.captures["function"]?.[0].node;
+        assertNonterminalNode(funcNode);
+        const funcDef = new FunctionDefinition(funcNode);
+        const name = funcDef.name.cst.unparse().trim();
```

Comment:
> cst.unparse().trim();

this is calling for a helper function (on our side, not saying to implement that here) 👀